packages/python/javdb_platform/db.py

```
@@ -1818,21 +2552,441 @@ def db_create_report_session(
+                        "Rollback step failed (table=%s action=%s id=%s): %s",
+                        main_table, action, target_id, e,
+                    )
+
+            if applied_ids:
+                # Tidy only the audit rows that replayed successfully. Any
+                # drifted tail remains for manual recovery.
+                for i in range(0, len(applied_ids), 900):
+                    chunk = applied_ids[i:i + 900]
+                    placeholders = ', '.join('?' for _ in chunk)
+                    conn.execute(
+                        f"DELETE FROM {audit_table} WHERE Id IN ({placeholders})",
```

Ruff (0.15.12)

[error] Possible SQL injection vector through string-based query construction

(S608)